### PR TITLE
SWIFT-373: Introduce Operation protocol, CountOperation, and DistinctOperation 

### PR DESCRIPTION
**Pull request by kmahar**
Part 1 of many. This introduces the `Operation` protocol and migrates the non-cursor `MongoCollection` "read" methods to use it.

### CODE REVIEW

File: Sources/MongoSwift/Operations/CountOperation.swift [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import mongoc
+
+/// An operation corresponding to a "count" command on a collection.
+internal struct CountOperation<T: Codable>: Operation {
+    private let collection: MongoCollection<T>
+    private let filter: Document
```
Comment (kmahar) [inline Sources/MongoSwift/Operations/CountOperation.swift]: My one question on this PR is how much we care to duplicate documentation on what these properties are. They are already all documented on the `count` method... is it worth pasting that here too? (same q applies to distinct.)

Comment (mbroadst) [reply Sources/MongoSwift/Operations/CountOperation.swift]: as long as it doesn't mess with your jazzy count, I don't see a compelling reason to duplicate options here

File: Sources/MongoSwift/Operations/CountOperation.swift [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            self.collection._collection, MONGOC_QUERY_NONE, self.filter.data, 0, 0, opts?.data, rp, &error)
+
+        if count == -1 { throw parseMongocError(error) }
+
+        return Int(count)
+    }
+}
+
+/// Options to use when executing a `count` command on a `MongoCollection`.
+public struct CountOptions: Encodable {
```
Comment (mbroadst) [inline Sources/MongoSwift/Operations/CountOperation.swift]: [nit]: you use the `CountOptions` in the operation above, consider moving the definition before the operation definition.

File: Sources/MongoSwift/Operations/Operation.swift [hunk 3]
```diff
@@ -0,0 +1,8 @@
+/// A protocol for operation types to conform to. An `Operation` instance corresponds to any single operation a user
```
Comment (mbroadst) [inline Sources/MongoSwift/Operations/Operation.swift]: I don't remember what the rules are for comments, but I think y'all moved to c-style comments at a certain point?

File: Sources/MongoSwift/Operations/CountOperation.swift (hunk 1 above)
Comment (kmahar) [reply Sources/MongoSwift/Operations/CountOperation.swift]: it's all private/internal, so jazzy won't care.

File: Sources/MongoSwift/Operations/Operation.swift (hunk 3 above)
Comment (kmahar) [reply Sources/MongoSwift/Operations/Operation.swift]: we standardized on (from development guide):
> We use C-style documentation blocks (`/** ... */`) for documentation longer than 3 lines, and triple-slash (`///`) for shorter documentation.

Comment (patrickfreed) [reply Sources/MongoSwift/Operations/Operation.swift]: I've been doing single and double line docstrings as `///` and c-style for everything longer.

File: Sources/MongoSwift/Operations/CountOperation.swift (hunk 2 above)
Comment (kmahar) [reply Sources/MongoSwift/Operations/CountOperation.swift]: done here and for distinct

**Pull request merged**